elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -702,11 +704,18 @@ public PersistentResource getRelation(String relation, String id) {
      * @return collection relation
      */
     public Set<PersistentResource> getRelation(String relationName) {
-        if (requestScope.getTransaction() != null && !requestScope.getPredicates().isEmpty()) {
-            final Class<?> entityClass = dictionary.getParameterizedType(obj, relationName);
-            final String valType = dictionary.getBinding(entityClass);
-            final Set<Predicate> filters = new HashSet<>(requestScope.getPredicatesOfType(valType));
-            return getRelation(relationName, filters);
+        // if pagination, then sorting with filters...
+        boolean hasFilters = !requestScope.getPredicates().isEmpty();
+        Set<Predicate> filters = null;
+        if (requestScope.getTransaction() != null && requestScope.getQueryParams().isPresent()) {
```

Comment:
This feels to me like it could be a trap for future query param use cases. I feel like there should be a way for us to check to see if this is a get request with query params that has a transaction (eg. bound to data store transactions).